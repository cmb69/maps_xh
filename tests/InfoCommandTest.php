<?php

namespace Maps;

use ApprovalTests\Approvals;
use org\bovigo\vfs\vfsStream;
use PHPUnit\Framework\TestCase;
use Plib\DocumentStore2 as DocumentStore;
use Plib\FakeSystemChecker;
use Plib\SystemChecker;
use Plib\View;

class InfoCommandTest extends TestCase
{
    private DocumentStore $store;
    private SystemChecker $systemChecker;
    private View $view;

    public function setUp(): void
    {
        vfsStream::setup("root");
        $this->store = new DocumentStore(vfsStream::url("root/"));
        $this->systemChecker = new FakeSystemChecker();
        $this->view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["maps"]);
    }

    private function sut(): InfoCommand
    {
        return new InfoCommand("./", $this->store, $this->systemChecker, $this->view);
    }

    public function testRendersPluginInfo(): void
    {
        $response = $this->sut()();
        $this->assertSame("Maps 1.0-dev", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
