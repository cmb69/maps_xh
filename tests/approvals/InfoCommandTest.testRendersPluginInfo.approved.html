
<article class="maps_plugininfo">
  <h1>Maps 1.0-dev</h1>
  <section class="maps_syscheck">
    <h2>System Check<h2>
    <p class="xh_fail">PHP version ≥ 8.0.0: bad</p>
    <p class="xh_fail">CMSimple_XH version ≥ 1.7.0: bad</p>
    <p class="xh_fail">Plib_XH version ≥ 1.9: bad</p>
    <p class="xh_fail">./config/ is writable: bad</p>
    <p class="xh_fail">./css/ is writable: bad</p>
    <p class="xh_fail">./languages/ is writable: bad</p>
    <p class="xh_fail">vfs://root/ is writable: bad</p>
  </section>
</article>
